src/V3Randomize.cpp

```diff
@@ -625,8 +626,15 @@ class ConstraintExprVisitor final : public VNVisitor {
             varRefp->classOrPackagep(classOrPackagep);
             methodp->addPinsp(varRefp);
             size_t width = varp->width();
-            if (VN_IS(varp->dtypep(), DynArrayDType) || VN_IS(varp->dtypep(), QueueDType))
-                width = varp->dtypep()->subDTypep()->width();
+            AstNodeDType* tmpDtypep = varp->dtypep();
+            if (VN_IS(tmpDtypep, UnpackArrayDType) || VN_IS(tmpDtypep, DynArrayDType)
+                || VN_IS(tmpDtypep, QueueDType)) {
+                while (VN_IS(tmpDtypep, UnpackArrayDType) || VN_IS(tmpDtypep, DynArrayDType)
+                       || VN_IS(tmpDtypep, QueueDType)) {
+                    tmpDtypep = tmpDtypep->subDTypep();
+                }
+                width = tmpDtypep->width();
+            }
```

Comment:
```suggestion
            AstNodeDType* tmpDtypep = varp->dtypep();
            while (VN_IS(tmpDtypep, UnpackArrayDType) || VN_IS(tmpDtypep, DynArrayDType)
                || VN_IS(tmpDtypep, QueueDType)) tmpDtypep = tmpDtypep->subDTypep();
            size_t width = tmpDtypep->width();
```

Note this works because varp->width() == varp->dtypep()->width().